// react-native-linear-gradient-text/lib/commonjs/components/LinearGradientText.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LinearGradientText = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _maskedView = _interopRequireDefault(require("@react-native-masked-view/masked-view"));

var _reactNativeLinearGradient = _interopRequireDefault(require("react-native-linear-gradient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const LinearGradientText = props => {
  const {
    text,
    textStyle = {},
    colors,
    start = {
      x: 0,
      y: 1
    },
    end = {
      x: 1,
      y: 1
    }
  } = props;
  return /*#__PURE__*/_react.default.createElement(_maskedView.default, {
    maskElement: /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
      style: [styles.maskText, textStyle]
    }, text)
  }, /*#__PURE__*/_react.default.createElement(_reactNativeLinearGradient.default, {
    colors: colors,
    start: start,
    end: end
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Text, {
    style: [styles.text, textStyle]
  }, text)));
};

exports.LinearGradientText = LinearGradientText;

const styles = _reactNative.StyleSheet.create({
  maskText: {
    backgroundColor: 'transparent'
  },
  text: {
    opacity: 0
  }
});
